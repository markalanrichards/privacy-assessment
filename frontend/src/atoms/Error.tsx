import * as React from "react";

export interface AtomProps {
  message: string;
  title: string;
}

const Display: React.SFC<AtomProps> = ({ title, message }) => (
  <div>
    <h3>{title}</h3>
    <span>{message}</span>
  </div>
);

export const Error = ({ graphQLErrors, networkError, extraInfo, message }) => (
  <div>
    {/* {console.log({ graphQLErrors, networkError, extraInfo, message })} */}
    <h1>AHhh Error</h1>
    {graphQLErrors
      ? graphQLErrors.map((props, i) => (
          <Display key={i} title="graphQLErrors" {...props} />
        ))
      : null}
    {networkError
      ? networkError.map((props, i) => (
          <Display key={i} title="networkError" {...props} />
        ))
      : null}
    {extraInfo
      ? extraInfo.map((props, i) => (
          <Display key={i} title="extraInfo" {...props} />
        ))
      : null}
    {message ? <Display title="message" message={message} /> : null}
  </div>
);
