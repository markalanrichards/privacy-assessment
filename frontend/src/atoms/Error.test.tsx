import * as React from "react";
import { shallow, mount } from "enzyme";
import { Error } from "./Error";

test("error - loads no message", () => {
  const mockData = {
    graphQLErrors: null,
    networkError: null,
    extraInfo: null,
    message: null
  };
  const wrapper = shallow(<Error {...mockData} />);
  expect(wrapper.text()).toBe("AHhh Error");
  expect(wrapper).toMatchSnapshot();
});

test("error - graphQLErrors", () => {
  const mockData = {
    graphQLErrors: [{ data: "error" }],
    networkError: null,
    extraInfo: null,
    message: null
  };
  const wrapper = mount(<Error {...mockData} />).find("h3");

  expect(wrapper.text()).toBe("graphQLErrors");
  expect(wrapper).toMatchSnapshot();
});

test("error - networkError", () => {
  const mockData = {
    graphQLErrors: null,
    networkError: [{ data: "error" }],
    extraInfo: null,
    message: null
  };
  const wrapper = mount(<Error {...mockData} />).find("h3");

  expect(wrapper.text()).toBe("networkError");
  expect(wrapper).toMatchSnapshot();
});

test("error - extraInfo", () => {
  const mockData = {
    graphQLErrors: null,
    networkError: null,
    extraInfo: [{ data: "error" }],
    message: null
  };
  const wrapper = mount(<Error {...mockData} />).find("h3");

  expect(wrapper.text()).toBe("extraInfo");
  expect(wrapper).toMatchSnapshot();
});

test("error - message", () => {
  const mockData = {
    graphQLErrors: null,
    networkError: null,
    extraInfo: null,
    message: "error message"
  };
  const wrapper = mount(<Error {...mockData} />);

  expect(wrapper.find("h3").text()).toBe("message");
  expect(wrapper.find("span").text()).toBe("error message");
  expect(wrapper).toMatchSnapshot();
});
