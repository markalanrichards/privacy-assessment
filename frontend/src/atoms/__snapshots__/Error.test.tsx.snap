// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error - extraInfo 1`] = `
<h3>
  extraInfo
</h3>
`;

exports[`error - graphQLErrors 1`] = `
<h3>
  graphQLErrors
</h3>
`;

exports[`error - loads no message 1`] = `
<div>
  <h1>
    AHhh Error
  </h1>
</div>
`;

exports[`error - message 1`] = `
<Component
  extraInfo={null}
  graphQLErrors={null}
  message="error message"
  networkError={null}
>
  <div>
    <h1>
      AHhh Error
    </h1>
    <Display
      message="error message"
      title="message"
    >
      <div>
        <h3>
          message
        </h3>
        <span>
          error message
        </span>
      </div>
    </Display>
  </div>
</Component>
`;

exports[`error - networkError 1`] = `
<h3>
  networkError
</h3>
`;
